Narayana JTA: do not allow to use datasource XA transactions with JDBC object store

michalvavrik opened the pull request:
closes: #33715

Changes in Narayana JTA integration module are down to the fact that `quarkus.datasource.test.jdbc.transactions` is build time property (and so is db-kind), so they can't be changed for integration test in test profile. It needs to be set in `application.properties` to disabled in order to use it as JDBC object store.

michalvavrik committed:
Prohibit using Narayana JDBC object store with Agroal XA

gsmet reviewed (changes requested):
Thanks!

I'm not completely convinced by your approach of reading the configuration this way. I think we should add more info in the build item instead.

I will have a look tomorrow and add a commit to discuss this with you as it's going to be easier with some code.

michalvavrik commented:
Sure @gsmet , thanks.

zhfeng reviewed (approved):
Thanks @michalvavrik and these changes look good to me!. It could be better if we can find a easier way to read the configurations at build time.

gsmet committed:
Use the source of truth to publish JdbcDataSourceBuildItem

Until now, we were using some transformed data, which wasn't very
practical when we needed more information.

gsmet committed:
Simplify resolution of the config by pushing the info to the build item

gsmet commented:
@michalvavrik I rebased and pushed 2 additional commits. Let me know what you think.

(I haven't run all the tests, only those of Agroal and Narayana, we will see how it goes :))

michalvavrik commented:
I agree this is better than my version, I only mentioned one unused method parameter (please see comment). Tell me how to fix it (if you want to resolve it yourself, please do). Glad for your help.

gsmet commented:
Ah, yes, I'll fix it later today.

gsmet commented:
@michalvavrik sorry for the delay, I fixed the remaining issue, rebased and force pushed.

michalvavrik commented:
@gsmet I think OTEL failure is related, I am busy today, but I will look into it tomorrow unless you solve it first.

gsmet commented:
I don't see any dependency to Narayana or Datasource in this test but there might be something given it was triggered by GIB.

I have no idea how it could be related though. I will be busy on other things too so I let you have a look when you have some time. There's nothing pressing that requires you to have a look tomorrow, you can have a look on Monday.

gsmet commented:
It's failing here too: https://github.com/quarkusio/quarkus/pull/35067 so it's not related, I think we can merge.
